StackExchange.Redis/StackExchange/Redis/GeoEntry.cs

```diff
@@ -0,0 +1,284 @@
+    /// </summary>
+    [Flags]
+    public enum GeoRadiusOptions
+    {
+        /// <summary>
+        /// No Options
+        /// </summary>
+        None = 0,
+        /// <summary>
+        /// Redis will return the coordinates of any results.
+        /// </summary>
+        WITHCOORD = 1,
+        /// <summary>
+        /// Redis will return the distance from center for all results.
+        /// </summary>
+        WITHDIST = 2,
+        /// <summary>
+        /// Redis will return the geo hash value as an integer. (This is the score in the sorted set)
+        /// </summary>
+        WITHHASH = 4
+    }
+
+    /// <summary>
+    /// The result of a GeoRadius command.
+    /// </summary>
+    public struct GeoRadiusResult
+    {
+        internal readonly RedisValue member;
+        internal readonly GeoRadius geoRadius;
+        internal readonly double? distanceFromCenter;
+        internal readonly long? geoHash;
+        internal readonly GeoPosition? geoPosition;
+
+        /// <summary>
+        /// The matched member.
+        /// </summary>
+        public RedisValue Member => member;
+        /// <summary>
+        /// The original GeoRadius command
+        /// </summary>
+        public GeoRadius GeoRadius => geoRadius;
+        /// <summary>
+        /// The distance of the matched member from the center of the geo radius command.
+        /// </summary>
+        public double? DistanceFromCenter => distanceFromCenter;
+        /// <summary>
+        /// The Geo Hash of the matched member as an integer. (The key in the sorted set)
+        /// </summary>
+        public long? GeoHash => geoHash;
+        /// <summary>
+        /// The coordinates of the matched member.
+        /// </summary>
+        public GeoPosition? GeoPosition => geoPosition;
+
+        /// <summary>
+        /// Returns a new GeoRadiusResult
+        /// </summary>
+        public GeoRadiusResult(RedisValue member,GeoRadius geoRadius,double? distanceFromCenter,long? geoHash,GeoPosition? geoPosition)
+        {
+            this.member = member;
+            this.geoRadius = geoRadius;
+            this.distanceFromCenter = distanceFromCenter;
+            this.geoHash = geoHash;
+            this.geoPosition = geoPosition;
+        }
+
+    }
+
+    /// <summary>
+    /// Represents a GeoRadius command and its options.
+    /// </summary>
+    public class GeoRadius
+    {
+        internal readonly GeoUnit geoUnit;
+        internal readonly GeoRadiusOptions geoRadiusOptions ;
+        internal readonly int maxReturnCount;
+        internal readonly RedisKey key;
+        internal readonly GeoPosition geoPosition;
+        internal readonly double radius;
+
+        /// <summary>
+        /// The Radius size of this GeoRadius command.
+        /// </summary>
+        public double Radius => radius;
+        /// <summary>
+        /// The center point to base the search.
+        /// </summary>
+        public GeoPosition GeoPosition => geoPosition;
```

Comment:
but what *is* the position? is this the center? if so: `Center` or `Origin` might be a better name.